src/NHibernate.Test/NHSpecificTest/NH3848/HqlTestFixture.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using NHibernate.Transform;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH3848
+{
+	public class HqlTestFixture : TestFixture
+	{
+		[Ignore("We can't write such query using hql, because we can't use 'with' clause when we're fetching collection.")]
+		public override void ChildCollectionsFromLeftOuterJoinWithOnClauseRestrictionOnCollectionShouldNotBeInSecondLevelCache()
+		{
+			throw new NotImplementedException();
+		}
+
+		protected override IList<Customer> GetCustomersWithFetchedOrdersWithoutRestrictions(ISession session)
+		{
+			var query = session.CreateQuery("from Customer as c left join fetch c.Orders as o");
+			query.SetResultTransformer(new DistinctRootEntityResultTransformer());
+			return query.List<Customer>();
+		}
+
+		protected override IList<Customer> GetCustomersWithOrdersEagerLoaded(ISession session)
+		{
+			var query = session.CreateQuery("from Customer as c left join fetch c.Orders as o");
+			query.SetResultTransformer(new DistinctRootEntityResultTransformer());
+			return query.List<Customer>();
+		}
+
+		//We can't write such query using hql, because we can't use 'with' clause when we're fetching collection.
+		protected override IList<Customer> GetCustomersByOrderNumberUsingOnClause(ISession session, int orderNumber)
+		{
+			throw new NotImplementedException();
+		}
+
+		protected override IList<Customer> GetCustomersByOrderNumberUsingWhereClause(ISession session, int orderNumber)
+		{
+			var numberParameterName = "number";
+			var query = session.CreateQuery(string.Format("from Customer as c " +
+														   "left join fetch c.Orders as o " +
+														   "where o.Number = :{0}", numberParameterName));
```

Comment:
A formatting like
```c#
var query =
	session.CreateQuery(
		string.Format(
			"from Customer as c " +
			"left join fetch c.Orders as o " +
			"where o.Number = :{0}",
			numberParameterName));
```
Would be more readable.

But in fact I would rather write it as:
```c#
var query =
	session.CreateQuery(
		"from Customer as c " +
		"left join fetch c.Orders as o " +
		$"where o.Number = :{numberParameterName}");
```

Or even:		
```c#
var query =
	session.CreateQuery(
		"from Customer as c " +
		"left join fetch c.Orders as o " +
		"where o.Number = :number");
```
To keep it simple.

And do the same for following queries.